### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by asvetlov**

### CODE REVIEW

**Comment by samuelcolvin** [comment]
Great idea. Let me know if you want any help.

**Comment by asvetlov** [comment]
I'd like to get an initial working setup and iterate over in separate requests.
40 min per PR for appveyor is too much

**Comment by webknjaz** [comment]
Here's a very good curated template for Azure Pipelines: https://github.com/tox-dev/azure-pipelines-template

**Comment by webknjaz** [comment]
But yes, we can merge whatever works initially and iterate on that later.

File: tests/test_connector.py [hunk 1]
```diff
@@ -56,7 +56,15 @@ def unix_server(loop, unix_sockname):
...
         await runner.setup()
         site = web.UnixSite(runner, unix_sockname)
-        await site.start()
+        try:
+            await site.start()
+        except OSError as exc:
+            if str(exc) == "AF_UNIX path too long":
+                if sys.platform == "linux":
+                    raise
+                pytest.skip("Some UNIXes has too strict limitation "
```
Comment (webknjaz) [inline tests/test_connector.py]: Actually all of them. It just happens that the default tmp path prefixed there under macOS is longer (`/private/tmp/...` vs. `/tmp/...`). I think, the right way to fix it is to substitute the temp dir.
It should be possible to do `export TMPDIR=/tmp` in CI. Try it.


Ref: https://unix.stackexchange.com/a/367012/27133

Comment (webknjaz) [reply tests/test_connector.py]: Also, in this case, it should be probably `xfail()` instead of `skip()` if you still want to keep it. But I don't like the idea of skipping tests when it's possible to address this properly.

Comment (webknjaz) [inline tests/test_connector.py]: It looks like a proper workaround it using `tmpdir_factory` fixture: https://github.com/agronholm/anyio/pull/42/files#diff-f0c1c7497545019b8489fa8bc06be6cdR208

Comment (webknjaz) [inline tests/test_connector.py]: Ref: #3572

Comment (asvetlov) [reply tests/test_connector.py]: xfail sounds good!

Comment (asvetlov) [reply tests/test_connector.py]: It is a workaround that doesn't guarantee short path anyway :(
Also, I prefer `tmp_path_factory` over `tmpdir_factory`

Comment (webknjaz) [inline tests/test_connector.py]: Once this https://github.com/aio-libs/aiohttp/pull/3832 is merged, revert the hack.

Comment (webknjaz) [reply tests/test_connector.py]: Yep, using `tmp_path_factory` now.

Comment (webknjaz) [reply tests/test_connector.py]: I've added a few improvements and it is now guaranteed to be short.

File: tests/test_web_runner.py [hunk 2]
```diff
@@ -110,7 +111,15 @@ def test_non_app() -> None:
     await tcp.start()
     path = str(tmpdir / 'tmp.sock')
```
Comment (webknjaz) [inline tests/test_web_runner.py]: This will have to use a proper sock from a fixture.

File: setup.py [hunk 3]
```diff
@@ -3,9 +3,6 @@
 import pathlib
 import re
 import sys
-from distutils.command.build_ext import build_ext
```
Comment (webknjaz) [inline setup.py]: Can this be moved to a separate PR?

File: tests/test_client_functional.py [hunk 4]
```diff
@@ -1,5 +1,3 @@
-"""HTTP client functional tests against aiohttp.web server"""
```
Comment (webknjaz) [inline tests/test_client_functional.py:1]: Please keep the module docstring.

File: tests/test_client_functional.py [hunk 5]
```diff
@@ -1200,7 +1198,7 @@ def do_release():
     app.router.add_post('/', handler)
     client = await aiohttp_client(app)
 
-    with fname.open() as f:
```
Comment (webknjaz) [inline tests/test_client_functional.py:1201]: It'd be nice to move this to a separate PR.

File: tests/test_client_functional.py (hunk 4 above)
Comment (asvetlov) [reply tests/test_client_functional.py:1]: Why? What is the point?

File: tests/test_client_functional.py (hunk 5 above)
Comment (asvetlov) [reply tests/test_client_functional.py:1201]: The test fails on azure but is passed on appveyor

File: setup.py (hunk 3 above)
Comment (asvetlov) [reply setup.py]: Yes, sure

File: tests/test_client_functional.py (hunk 4 above)
Comment (webknjaz) [reply tests/test_client_functional.py:1]: When I open a python module, regardless of whether it is a test suite or not it helps me to understand what I'm about to see next, get some context and decide whether I need to read further.

File: tests/test_client_functional.py (hunk 5 above)
Comment (webknjaz) [reply tests/test_client_functional.py:1201]: Okay, but I still think that it'd be nice to separate this and only keep the CI config here so that it'd be more atomic.

File: tests/test_client_functional.py (hunk 4 above)
Comment (asvetlov) [reply tests/test_client_functional.py:1]: I have a different position: docstrings are for public API only.
It should be just a comment otherwise.

File: tests/test_client_functional.py (hunk 5 above)
Comment (asvetlov) [reply tests/test_client_functional.py:1201]: I cannot test it in a separate PR, this is the problem.

**Comment by webknjaz** [comment]
I think it may make sense to focus on GitHub workflows as they provide more resources.

**Comment by asvetlov** [comment]
Like Windows and MacOS boxes?

**Comment by webknjaz** [comment]
Yes

**Comment by asvetlov** [comment]
"GitHub Actions is currently in limited public beta and is subject to change. We strongly recommend that you do not use this feature for high-value workflows and content during the beta period."

I want to use a tool, not fight with unstable yet features. aiohttp itself provides a lot of space for experimenting.

**Comment by webknjaz** [comment]
I think they forgot to update that. Basically, anybody applying for access gets it instantly now and it's going to be generally available on Nov 13.
I've been playing with it for a while and I'm happy with the results.

**Comment by asvetlov** [comment]
Done by separate PRs
